### ISSUE TITLE
REGEX Blocklist as DNS Blocklist is incorrectly parsed.

### ISSUE THREAD

**Issue by CipherOps** [AdguardTeam/AdGuardHome#5015]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not ask a question

### Operating system type

Linux, Other (please mention the version in the description)

### CPU architecture

x86

### Installation

GitHub releases or script from README

### Setup

Other (please mention in the description)

### AdGuard Home version

On one machine, DHCP is handled by router

#### What did you do?
Added https://raw.githubusercontent.com/CipherOps/AdGuardBlocklists/main/REGEX.txt as DNS Blocklist
#### Expected result
REGEX filtering of domains
#### Actual result
Falsely blocking domains as REGEX
#### Screenshots (if applicable)
![image](https://user-images.githubusercontent.com/53319122/194684717-3a21b6ae-226c-42a1-8c3d-2b9410cbdfc1.png)

![image](https://user-images.githubusercontent.com/53319122/194684724-80b0c578-dbb1-426f-8eee-e063cef4eebf.png)

#### Additional information
Copy and pasting the REGEX as Custom Filter Rules works fine, no issues, added the link to DNS Blocklists, it blocks domains falsely. Checking REGEX rules against domains blocked with regex101.com shows the REGEX isn't matching the domain blocked, even though AdGuard Home said it was.

**Comment by CipherOps** [comment]
I do not think so, when I removed the domain as an adlist and just copy and pasted the contents into custom filtering rules, these domains were no longer blocked, however I have no way of knowing the cname they would've loaded from as it is not reported in the interface.

**Comment by ainar-g** [comment]
@CipherOps, you can check it by using `dig` or any other DNS lookup tool.  For example, I've made the following query:

```sh
dig IN A 'prod-video-cms-rt-microsoft-com.akamaized.net' @$MY_ADGUARD_HOME_IP
```

And the result was:

```none
…
;; ANSWER SECTION:
prod-video-cms-rt-microsoft-com.akamaized.net. 36 IN CNAME a1985.g2.akamai.net.
a1985.g2.akamai.net.    36      IN      A       213.155.157.97
a1985.g2.akamai.net.    36      IN      A       213.155.157.144
…
```

And the `.g2.` part here matches the regexp.

Note that CNAMEs can change over time, so it is possible that you moving the rules from one list to another didn't actually affect anything.

**Comment by kevindd992002** [comment]
I'm having the same issue where this blocklist is blocking login.yahoo.com:

![image](https://user-images.githubusercontent.com/3907957/196344362-ffa0e227-ee18-43fb-906d-be2c145a75d1.png)

![image](https://user-images.githubusercontent.com/3907957/196344401-3e03e197-4e32-4b64-9d90-1383194f5503.png)

It can't load the images in https://ph.shein.com/ too

**Comment by devipasigner** [comment]
Hi, @kevindd992002 , @CipherOps this is not an issue with AdGuardHome but rather an issue with your regex rules. I manually checked all of the domains reported here and guess what? All of the domains have a cname that contains “g[0-9]” with matches with the “/.*\.g[0-9]+\..*/” regex rule in your blocklist. Like @ainar-g pointed out, cnames CAN change over time so you moving the regex rules from blocklists to custom rules really didn’t change anything. My suggestion would be to remove that rule and see how it goes. This issue can be closed.
